### ISSUE TITLE
RSA keys in .ssh no longer works with ssh-agent

### ISSUE THREAD

**Issue by thomasjsn** [ohmyzsh/ohmyzsh#12743]
### Describe the bug

After #12741, my key `.ssh/id_rsa`, is no longer loaded. Looking at the PR, it seems to now require keys to be `OPENSSL` format, but my key is `RSA`.

### Steps to reproduce

1. Have an `RSA` key in `.ssh`
2. Start the ssh-agent
3. Observe that the `RSA` key is not loaded, meaning its password is requested every time it is used

### Expected behavior

`RSA` keys should also work, as before.

### OS / Linux distribution

Arch

### Zsh version

5.9

### Oh My Zsh version

master (d2d5155)

### Terminal emulator

rxvt-unicode

### If using WSL on Windows, which version of WSL

None

**Comment by mikhailde** [comment]
Hi!

Thanks for reporting this issue! I've identified the problem: the recent change to the ssh-agent plugin focused on OpenSSH keys and inadvertently broke compatibility with older RSA key formats like the one you're using.

I'm working on a fix that will add support for both OpenSSH and legacy RSA/DSA key formats (like -----BEGIN RSA PRIVATE KEY-----). This will ensure that your existing key works as expected, while still allowing users to benefit from the improved handling of OpenSSH keys.

The fix will be included in the next release of the plugin. In the meantime, you can temporarily revert to the previous version of the plugin if this is blocking your workflow. Let me know if you need help with that.

**Comment by mikhailde** [comment]
@thomasjsn Thanks for reporting this issue! You are correct, the previous change in #12741 inadvertently broke support for RSA keys in PEM format. 

I've addressed this problem in a new pull request: #12758. This PR uses a more robust method for identifying private keys that should correctly handle your RSA key. 

Please check out #12758 and let me know if it resolves the issue for you. You can test it by updating your local repository with the changes from that PR and then restarting your shell.

I apologize for the inconvenience caused by the previous change.
